WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class AliceClient : BackgroundService, IAsyncDisposable
+	{
+		private Guid AliceId { get; }
+		private Guid RoundId { get; }
+		private ArenaClient ArenaClient { get; }
+		public IEnumerable<ICoin> CoinsToRegister { get; }
+
+		public AliceClient(Guid aliceId, Guid roundId, ArenaClient arenaClient, IEnumerable<ICoin> coinsToRegister)
+		{
+			AliceId = aliceId;
+			RoundId = roundId;
+			ArenaClient = arenaClient;
+			CoinsToRegister = coinsToRegister;
+		}
+
+		protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+		{
+			// Input registration phase
+			do
+			{
+				await Task.Delay(TimeSpan.FromSeconds(30), stoppingToken).ConfigureAwait(false);
+				await ConfirmationAsync().ConfigureAwait(false);
+			}
+			while (!stoppingToken.IsCancellationRequested);
+		}
+
+		private async Task ConfirmationAsync()
+		{
+			var inputWeight = 4 * Constants.P2wpkhInputVirtualSize;
+			var inputRemainingWeights = new[] { (long)ArenaClient.ProtocolMaxWeightPerAlice - inputWeight };
+
+			var amountCredentials = ArenaClient.AmountCredentialClient.Credentials;
+
+			await ArenaClient.ConfirmConnectionAsync(
+				RoundId,
+				AliceId,
+				inputRemainingWeights,
+				amountCredentials.ZeroValue.Take(ArenaClient.ProtocolCredentialNumber),
+				CoinsToRegister.Select(c => (Money)c.Amount)
+				).ConfigureAwait(false);
+		}
+
+		public async Task UnConfirmationAsync()
+		{
+			throw new NotImplementedException();
+			// TODO: Add .RequestHandler.RemoveInputAsync() to ArenaClient.
+			// Logger.LogInfo($"Round ({RoundId}), Alice ({AliceId}): Unconfirmed connection.");
+		}
```

Comment:
removing an input is only allowed during input registration so i think this won't be necessary

Reply:
sorry, not sure what I was thinking.. obviously the client can still unregister while in the confirmation loop, so this makes sense

Reply:
Added the impl.